src/DynamoCore/Extensions/ExtensionManager.cs

```diff
@@ -139,6 +146,12 @@ public void Remove(IExtension extension)
                 Log(fullName + " extension cannot be disposed properly: " + ex.Message);
             }
 
+            if (extension is IExtensionStorageAccess storageAccess &&
+                storageAccessExtensions.Find(x => (x as IExtension).UniqueId == extension.UniqueId) is null)
```

Comment:
shouldn't that be not null? I may be coffee deprived.

Reply:
this is probably a case of me over complicating stuff, i wanted to follow the existing pattern of adding to the `extensions` list. The reason its checking if its null is to make sure its not already in the `storageAccessExtensions` List, so it only gets added if it isnt there already.

But as you mention a dict or hashset might be more suitable, this was mainly done to just follow the existing implementation.

Reply:
right - but isn't this the remove section? You'd want the extension to be present before removing it right?

Reply:
Oh right!! 🤦‍♂️

My bad, you are absolutely right!